### PR TITLE
insights: convert webapp to use insights gql for just in time search insights

### LINKED ISSUE #35306: insights: convert webapp just in time search insights to use insights graphql
As of #34149 the gql `insightsViews` now supports running just in time search insights.  This means that all LineChart based insights data can be pulled via gql and the webapp no longer needs to maintain the separate logic to execute these queries.   

Switching to using the gql enables just in time search insights to use the `InsightViewFilters` making that functionality available across all LineChart insight types.

### PR DESCRIPTION
**Pull request by chwarwick**
Switch Just In Time search insights to pull data from the insights graphql api.

closes #35306

## Test plan
All tests pass

Manual Tests

- Verified Search Insights for some and all repos load
- Verified Capture Group insights for some and all repos load
- Verified Creating/Editing of new insights 
- Verified Creating a new insight from a filter

 

## App preview:

- [Web](https://sg-web-cw-insights-jit-api.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-orfqubnzme.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/insights/core/backend/gql-backend/deserialization/create-insight-view.ts [hunk 1]
```diff
@@ -65,32 +64,21 @@ export const createInsightView = (insight: InsightViewNode): Insight => {
...
-                        context: appliedFilters.searchContexts?.[0] ?? '',
-                    },
-                }
-            }
+            const { presentation, appliedFilters } = insight
 
             return {
                 ...baseInsight,
-                executionType: InsightExecutionType.Runtime,
+                executionType: InsightExecutionType.Backend,
```
Comment (vovakulikov) [inline client/web/src/enterprise/insights/core/backend/gql-backend/deserialization/create-insight-view.ts:71]: I think this is perhaps outside of this PR scope but I think we don't need this separation Runtime vs Backend execution time anymore. FYI @unclejustin

**Review by vovakulikov — APPROVED** [review]
LGTM. It also seems that we can completely remove `BuiltInInsight.tsx` component and therefore this PR unblocks a lot of opportunities to simplify many places on the FE (and remove a lot of code from frontend around just in time insights) 

Thanks @chwarwick, fan of your work.

File: client/web/src/enterprise/insights/core/backend/gql-backend/deserialization/create-insight-view.ts (hunk 1 above)
Comment (chwarwick) [reply client/web/src/enterprise/insights/core/backend/gql-backend/deserialization/create-insight-view.ts:71]: I agree but was cautious about removing it, the scope had already expanded farther than I had expected it to.  I assume after the live preview pr that there is quite a bit of code that can be removed.

**Comment by chwarwick** [comment]
@vovakulikov I ran into an issue with [this test](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/integration/insights/create-insights.test.ts?L136).  Should I try to convert it or just remove it? It does not look like any other backend created insights are tested in this way.

**Comment by vovakulikov** [comment]
@chwarwick It's worth trying to fix this test/covert it. This kind of test tries to test the full flow of the creation UI actions. We indeed don't have similar tests for other types of insights but this is our problem I think, we should do more tests like this one. 

if you won't be able to fix this one in a quick way I think I or @unclejustin could help with it.

File: client/web/src/enterprise/insights/core/backend/gql-backend/deserialization/create-insight-view.ts (hunk 1 above)
Comment (leonore) [reply client/web/src/enterprise/insights/core/backend/gql-backend/deserialization/create-insight-view.ts:71]: @chwarwick is it worth adding an issue to cleanup the code that will no longer be used, so we don't forget about it?

**Pull request merged**